### PR TITLE
Replace discussion-forum XBlock

### PR DESCRIPTION
**Pull request by itsjeyd**
This PR removes the old, solutions-specific `DiscussionXBlock` and replaces it with the new `DiscussionXBlock` introduced upstream. To ensure backward compatibility, the `discussion-forum` entry point provided by the old block is aliased to the `discussion` entry point provided by the new block.

The PR is part of the Eucalyptus rebase for the solutions fork.

cf. [YONK-381](https://openedx.atlassian.net/browse/YONK-381)

**Test instructions**

Assuming you are able to run Studio and LMS on the parent branch (`ziafazal/fix-broken-features`), perform these steps to check if units using the existing inline discussion block keep working after transitioning to the new block:

1.  Install the solutions version of the inline discussion block:

    1.  Add `-e common/lib/xblock/discussion` to `requirements/edx/local.txt` (otherwise the old block won't be installed):

            ...        
            -e common/lib/xblock/discussion
            -e openedx/core/lib/xblock_builtin/xblock_discussion
            -e openedx/core/lib/xblock_builtin/xblock_discussion_course

    2.  In `common/lib/xblock/discussion/setup.py`, change the package name to `discussion-forum` (to avoid a name clash with the new inline discussion block):
        
        ```python
        setup(
            name='discussion-forum',
            # ...
        )
        ```

    3.  Run `install -r requirements/edx/local.txt`.

2.  Run `pip freeze | grep discussion`. Verify that it produces the following output:
    
        discussion-edx-platform-extensions==1.0.0
        discussion-forum-0.1
        xblock-discussion==0.1
        xblock-discussion-course==0.1

3.  Add `discussion-forum` to the list of advanced modules for a course.

4.  Create a new unit and select "Advanced" > "Discussion" to add an inline discussion block. Rendering will fail because the block depends on an HTML template that has moved to another location as part of the Eucalyptus rebase. This is expected since you haven't switched to the feature branch for this PR yet.

5.  Stop Studio and uninstall the old discussion block via `pip uninstall discussion-forum`.

6.  Run `pip freeze | grep discussion`. Verify that it produces the following output:
    
        discussion-edx-platform-extensions==1.0.0
        xblock-discussion==0.1
        xblock-discussion-course==0.1

7.  Run `export NO_PREREQ_INSTALL=1` to avoid re-installing the old inline discussion block.

8.  Switch to the feature branch for this PR (`itsjeyd/replace-discussion-forum`).

9.  Run `install -r requirements/edx/local.txt` to update the new inline discussion block, which aliases the `discussion-forum` entry to `discussion`.

10. Start Studio (using `--fast` option) and reload the unit that you added a "Discussion" block to earlier. Observe that you now get a sensible preview that you can modify via the "EDIT" dialog.

**Known issues**

* In the LMS, clicking "Expand discussion" results in modal dialog saying

  > The thread you selected has been deleted. Please select another thread.

  Based on console output, there is no server-side URL for handling AJAX request from `DiscussionThreadView.prototype.loadResponses`.

  This issue is unrelated to the changes introduced in this PR (they don't touch the code that powers the new inline discussion block). A note about fixing it has been added to [YONK-401](https://openedx.atlassian.net/browse/YONK-401).

### CODE REVIEW

File: openedx/core/lib/xblock_builtin/xblock_discussion/setup.py [hunk 1]
```diff
@@ -28,7 +28,8 @@ def package_data(pkg, root_list):
     ],
     entry_points={
         'xblock.v1': [
-            'discussion = xblock_discussion:DiscussionXBlock'
+            'discussion = xblock_discussion:DiscussionXBlock',
+            'discussion-forum = xblock_discussion:DiscussionXBlock'
         ]
```
Comment (pomegranited) [inline openedx/core/lib/xblock_builtin/xblock_discussion/setup.py]: @itsjeyd Please add a comment explaining why this alias is necessary.

**Comment by pomegranited** [comment]
@itsjeyd Your PR is looking good!  Just a couple of things noted below.

- [x] I tested this using the PR test instructions, and the results were as described.  However, I had to make one more amendment to `ziafazal/fix-broken-features` to get the old `discussion-forum` modules to load in Studio.  Perhaps because of your [recent merge](https://github.com/edx-solutions/edx-platform/pull/741)?

    ```diff
git diff common/lib/xblock/discussion/discussion_forum/__init__.py
-from .discussion_forum import DiscussionXBlock, DiscussionCourseXBlock
+from .discussion_forum import DiscussionXBlock
+from xblock_discussion_course import DiscussionCourseXBlock
```
- [ ] Automated tests are failing, but these may be because the base branch is in progress?
    * The [CircleCI tests are failing](https://circleci.com/gh/edx-solutions/edx-platform/956), though it looks like out of memory errors?
    * I get the following errors for `cms` and `lms` tests when I run `paver test_python`:

       ```
Traceback (most recent call last):
  File "./manage.py", line 112, in <module>
    startup.run()
  File "/edx/app/edxapp/edx-platform/cms/startup.py", line 63, in run
    disable_solutions_apps_signals()
  File "/edx/app/edxapp/edx-platform/cms/startup.py", line 122, in disable_solutions_apps_signals
    from edx_solutions_api_integration.test_utils import SignalDisconnectTestMixin
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/edx_solutions_api_integration/test_utils.py", line 21, in <module>
    from course_metadata.signals import (
ImportError: cannot import name course_publish_handler_in_course_metadata
```

- [x] I read through the code, and the removed files look correct.  Though I believe the following directories can also be removed, and are no longer present in `edx:master`.  I've tested my course in Studio and LMS without these folders, and with a full re-compile of the static assets, and everything worked fine.
    * `common/static/xblock/discussion/`
    * `common/templates/discussion/`
    * `common/templates/js/discussion/`
    
- [ ] Includes documentation - [one comment requested](https://github.com/edx-solutions/edx-platform/pull/742/files#r77112935).  The remaining changes are code removal, so no further documentation is required.

**Comment by itsjeyd** [comment]
Thanks for the thorough review, @pomegranited! I added a comment that explains the purpose of the alias.

> However, I had to make one more amendment to `ziafazal/fix-broken-features` to get the old `discussion-forum` modules to load in Studio. Perhaps because of your recent merge?

Sorry about that. I think I wrote the test instructions after I made the changes in https://github.com/edx-solutions/edx-platform/pull/742/commits/6077d4f5f20ff46e88fad9dbdf07a8403d950c76 (which should have been part of #741); so there were no more references to `DiscussionCourseXBlock` in `common/lib/xblock/discussion` that could cause issues.

> Automated tests are failing, but these may be because the base branch is in progress?

Yes, fixing the failing tests is an ongoing process that is part of the overall rebase. Most of the (unrelated) tests that failed in the [build](https://circleci.com/gh/edx-solutions/edx-platform/956) you checked were also failing in the most recent [build](https://circleci.com/gh/edx-solutions/edx-platform/937) for #741, and the one test that's different seems unrelated as well (it comes from the `edx_solutions_api_integration` app).

> I believe the following directories can also be removed, ...
>    * `common/static/xblock/discussion/`
>    * `common/templates/discussion/`
>    * `common/templates/js/discussion/`

`common/static/xblock/discussion/` contains files that are referenced by the course-wide discussion block, see e.g. [xblock_discussion_course.py#L45](https://github.com/open-craft/edx-platform/blob/itsjeyd/replace-discussion-forum/openedx/core/lib/xblock_builtin/xblock_discussion_course/xblock_discussion_course.py#L45) and [xblock_discussion_course.py#L70](https://github.com/open-craft/edx-platform/blob/itsjeyd/replace-discussion-forum/openedx/core/lib/xblock_builtin/xblock_discussion_course/xblock_discussion_course.py#L70). The same goes for `common/templates/discussion/` (see [xblock_discussion_course.py#L53](https://github.com/open-craft/edx-platform/blob/itsjeyd/replace-discussion-forum/openedx/core/lib/xblock_builtin/xblock_discussion_course/xblock_discussion_course.py#L53)).

Some or perhaps even all of those files might get moved to different locations as part of [YONK-382](https://openedx.atlassian.net/browse/YONK-382), and if that happens we might be able to get rid of the containing directories.

`thread-show.underscore, the only file that lives in `common/templates/js/discussion/`, is referenced by one of the templates in `common/templates/discussion/`, so I think it's better to keep it for now.

**Comment by pomegranited** [comment]
:+1: Thank you for answering my questions!

- [x] I tested this using the PR instructions, and it worked as described.
- [x] I read through the code.
- [x] Includes documentation.

**Comment by itsjeyd** [comment]
Thanks @pomegranited!

@ziafazal Following up on #741, this is PR takes the next step towards cleaning up discussion-related code: removing the solutions-specific `DiscussionXBlock` in favor of the `DiscussionXBlock` that was added to the master fork not too long ago. We'd like to merge it to your branch. Would you be able to provide a second review?

**Comment by ziafazal** [comment]
LGTM 👍

**Comment by itsjeyd** [comment]
@ziafazal Thanks! Merging now.

**Pull request merged**